rest/route/github_comments.go

```diff
@@ -0,0 +1,56 @@
+package route
+
+import (
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/google/go-github/v52/github"
+)
+
+type githubComments struct {
+	baseURL string
+}
+
+func newGithubComments(baseURL string) githubComments {
+	return githubComments{baseURL}
+}
+
+// existingPatches returns a comment that is posted when there are existing patches for a commit SHA and
+// they will not be aborted, so the PR context is of the existing rather than the new PR.
+func (g *githubComments) existingPatches(patches []patch.Patch) string {
+	return fmt.Sprintf("There is an existing patch(es) for this commit SHA:\n%s\n\nPlease note that the status that is posted is not in the context of this PR but rather the (latest) existing patch and that may affect some tests that may depend on the particular PR. If your tests do not rely on any PR-specific values (like base or head branch name) then your tests will report the same status. If you would like a patch to run in the context of this PR and abort the other(s), comment 'evergreen retry'.", g.getLinksForPRPatches(patches))
+}
+
+// overridingPR returns a comment that is posted when there are existing patches for a commit SHA and
+// they will be aborted, so the PR context is of the new PR rather than the existing PR.
+func (g *githubComments) overridingPR(patches []patch.Patch) string {
+	return fmt.Sprintf("There is an existing patch(es) for this commit SHA that will be aborted:\n%s\n\nThe status reported will be corresponding to this PR rather than the previous existing ones. If you would like a patch to run for another PR and to abort this one, comment 'evergreen retry' on the corresponding PR.", g.getLinksForPRPatches(patches))
+}
+
+// overridenPR returns a comment that is posted when a PR's patch is aborted (or changed) in
+// favor of anther patch in a different PR context.
+func (g *githubComments) overridenPR(pr *github.PullRequest) string {
```

Comment:
nit: Sorry, English is hard but this one specifically is "overridden", only "overriding" has a single "d". I think the correct combination is overridden + overriding.